oxenss/snode/service_node.cpp

```diff
@@ -618,7 +635,80 @@ void ServiceNode::on_swarm_update(block_update&& bu) {
     // initiate_peer_test();
 }
 
-void ServiceNode::update_swarms() {
+static void request_for_recent_block_hashes(
+        uint64_t height, server::OMQ& omq, std::map<uint64_t, std::string>& height_to_hashes) {
+    uint64_t start_height = height - TEST_BLOCKS_BUFFER - 2;
+    if (start_height > height) // Overflow
+        start_height = 0;
+
+    if (start_height >= height) // Early out if there are no heights to query
+        return;
+
+    // Construct the OMQ request that batches all the heights to grab
+    fmt::memory_buffer request_buffer;
+    fmt::format_to(std::back_inserter(request_buffer), "{{\"heights\": [");
+    for (uint64_t h = start_height; h < height; h++) {
+        fmt::format_to(
+                std::back_inserter(request_buffer),
+                "{}{}",
+                h,
+                h == (height - 1) ? "]}" : ",");
+    }
+
+    std::string request_cmd = fmt::to_string(request_buffer);
```

Comment:
We should not be manually constructing json when we have a perfectly usable json library available, especially when not in performance critical code.  (I do realize the old code was doing a string construction, but if you're going to rewrite it, make it better instead of merely writing a better implementation of current bad idea).
```suggestion
    std::vector hr(height - start_height + 1, 0);
    std::iota(hr.begin(), hr.end(), start_height);
    std::string request_cmd = nlohmann::json{{"heights", hr}}.dump();
```
(or various other approaches; but the point is that the above removes all logic involving json formatting)

Reply:
Sure, I have addressed this in the comment further below by dropping the testing code which is what necessitated the hash retrieval.